dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,246 @@
+            return None
+        query = " or ".join(
+            "'{}' in parents".format(parent_id) for parent_id in parents_ids
+        )
+
+        query += " and trashed=false and title='{}'".format(name)
+
+        list_request = RequestListFile(self.drive, query)
+        item_list = self.execute_request(list_request)
+        return next(iter(item_list), None)
+
+    def resolve_remote_file(self, parents_ids, path_parts, create):
+        for path_part in path_parts:
+            item = self.get_drive_item(path_part, parents_ids)
+            if not item and create:
+                item = self.create_drive_item(parents_ids[0], path_part)
+            elif not item:
+                return None
+            parents_ids = [item["id"]]
+        return item
+
+    def subtract_root_path(self, parts):
+        if not hasattr(self, "root_id"):
+            return parts, [self.path_info.netloc]
+
+        for part in self.path_info.path.split("/"):
+            if parts and parts[0] == part:
+                parts.pop(0)
+            else:
+                break
+        return parts, [self.root_id]
+
+    def get_path_id_from_cache(self, path_info):
+        files_ids = []
+        parts, parents_ids = self.subtract_root_path(path_info.path.split("/"))
+        if (
+            path_info != self.path_info
+            and parts
+            and (parts[0] in self.cached_dirs)
+        ):
+            parents_ids = self.cached_dirs[parts[0]]
+            files_ids = self.cached_dirs[parts[0]]
+            parts.pop(0)
+
+        return files_ids, parents_ids, parts
+
+    def get_path_id(self, path_info, create=False):
+        files_ids, parents_ids, parts = self.get_path_id_from_cache(path_info)
+
+        if not parts and files_ids:
+            return files_ids[0]
+
+        file1 = self.resolve_remote_file(parents_ids, parts, create)
+        return file1["id"] if file1 else ""
+
+    def exists(self, path_info):
+        return self.get_path_id(path_info) != ""
+
+    def _upload(self, from_file, to_info, name, no_progress_bar):
+        dirname = to_info.parent
+        if dirname:
+            parent_id = self.get_path_id(dirname, True)
+        else:
+            parent_id = to_info.netloc
+
+        upload_request = RequestUploadFile(
+            {
+                "drive": self.drive,
+                "title": to_info.name,
+                "parent_id": parent_id,
+                "mime_type": "",
+            },
+            no_progress_bar,
+            from_file,
+            name,
+        )
+        self.execute_request(upload_request)
+
+    def _download(self, from_info, to_file, name, no_progress_bar):
+        file_id = self.get_path_id(from_info)
+        download_request = RequestDownloadFile(
+            {
+                "drive": self.drive,
+                "file_id": file_id,
+                "to_file": to_file,
+                "progress_name": name,
+                "no_progress_bar": no_progress_bar,
```

Comment:
minor: I'm not if it's consistent with other places we have, but it's really better to use `progress_bar` var to avoid double negations like this: `if not no_progress_bar` everywhere which are considered bad style and harder to read cc @efiop